core/templates/pages/exploration-player-page/layout-directives/audio-bar.directive.ts

```diff
@@ -77,10 +79,12 @@ angular.module('oppia').directive('audioBar', [
           };
 
           $scope.expandAudioBar = function() {
+            $rootScope.$broadcast('audioBarIsExpanded');
```

Comment:
Please find a way that doesn't use $rootScope -- we're getting rid of it as part of the angular migration.

In addition, using $rootScope is not a great practice anyway because it functions like a goto.

Reply:
Yeah, but for this we need kind of a shared service which is easily present in Angular but this $rootScope is required here since communication is taking place between two directives.

Reply:
We are getting rid of $rootScope. You might want to talk with @bansalnitish about alternatives, but one idea would be to try and pass the state between the directives via attributes. Or use a common service.

Reply:
Considering the use case, it's a child to parent communication and $rootScope is the best alternative, A common service would be overkill though we can use AudioPlayerService but that will be out of context(but can be done).
@bansalnitish  Would it be a problem to increase the use of rootScope?

Reply:
I'm still in the stage of finding alternatives to eliminate the use of $rootScope logically in the code base. Can't we pass the state between the directives via attributes as Sean suggested? If we can't then I cannot think of any other alternative rather than creating a common service. Though, creation of common service would lead to the fact that we are not using $rootScope explicitly (though we are using it logically).

Reply:
Fixed. Thanks